src/routes/InventoryDetail.js

```diff
@@ -98,30 +108,37 @@ const Inventory = () => {
   const store = useStore();
   const navigate = useNavigate();
   const dispatch = useDispatch();
-
+  const [availableApps, setAvailableApps] = useState([]);
   const entityLoaded = useSelector(
     ({ entityDetails }) => entityDetails?.loaded
   );
   const entity = useSelector(({ entityDetails }) => entityDetails?.entity);
   const { cloud_provider: cloudProvider, host_type: hostType } = useSelector(
     ({ systemProfileStore }) => systemProfileStore?.systemProfile || []
   );
-  const availableApps = useMemo(
-    () =>
-      appList.map((app) =>
-        app.name === 'ros'
-          ? {
-              ...app,
-              isVisible: cloudProvider === 'aws',
-            }
-          : app
-      ),
-    [cloudProvider]
-  );
+  useEffect(() => {
+    let osSlug = entity?.system_profile?.operating_system?.name
+      .replace(' ', '-')
+      .toUpperCase();
+
+    let newApps =
+      entity &&
+      appList[osSlug]?.map((app) => {
+        app.isDisabled = app.nonEdge && hostType === 'edge' ? true : false;
+        app['data-cy'] = `${app.name}-tab`;
+        app.name === 'ros' && {
+          ...app,
+          isVisible: cloudProvider === 'aws',
+        };
+        return app;
+      });
+
+    setAvailableApps(newApps);
+  }, [entity, cloudProvider]);
```

Comment:
An idea: Can we move this set of code to decide the new apps list out of the hook? Because the entity should not change when the user is viewing the page, also cloudProvider does not change consequently. They change only when the user goes to another page or views another host detail intentionally, leaving no necessity for the hook. Moving the code out of the hook gives more speed as we will not have to set a state change

Reply:
On page load the entity takes a bit to come in, initially without the osSlug we could have done that, but now that we wait on entity to decide what apps are displayed, this is necessary